### PR TITLE
Automate PyPi uploads on release

### LINKED ISSUE #493: Automate version bump on release
Create workflow/action to trigger both a change in version and a new GitHub release together (which in turn triggers a deploy to PyPI).

### PR DESCRIPTION
**Pull request by AdamGleave**
Fixes #493 via using `setuptools_scm`

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -92,8 +92,9 @@ def run(self):
         "tqdm",
         "scikit-learn>=0.21.2",
         "stable-baselines3>=1.5.0",
-        # TODO(nora) switch back to PyPi once 0.8.3 makes it to release:
-        "sacred@git+https://github.com/IDSIA/sacred.git@0.8.3",
+        # TODO(adam) switch to upstream release if they make it
+        #  See https://github.com/IDSIA/sacred/issues/879
+        "chai-sacred>=0.8.3",
```
Comment (Rocamonde) [inline setup.py:97]: Lol this is bold but effective for now. It is a bit worrisome that the devs have not been updating the release on PyPI regularly.

**Comment by Rocamonde** [comment]
LGTM. Now we'd have to create a release draft, and make sure to add useful information in the release notes

**Review by norabelrose — CHANGES_REQUESTED** [review]
This looks good to me except that the docs seem to be failing to build in the CI because the readthedocs build is still using Python 3.7. This PR uses `importlib.metadata` which was added in Python 3.8. We should probably change `readthedocs.yml` to Python 3.8 to fix this?

**Comment by Rocamonde** [comment]
> This looks good to me except that the docs seem to be failing to build in the CI because the readthedocs build is still using Python 3.7. This PR uses `importlib.metadata` which was added in Python 3.8. We should probably change `readthedocs.yml` to Python 3.8 to fix this?

This is done in #484, but this should probably get merged first, so I agree with this.

**Comment by AdamGleave** [comment]
Fixed doc building issue now by bumping to 3.8.

**Comment by AdamGleave** [comment]
Windows tests failing are flaky and will be fixed in https://github.com/DLR-RM/stable-baselines3/pull/979 -- we were previously hotfixing to that, but PyPI won't allow dependencies on specific GH issues.

I'm going to just skip these tests on Windows for now as a workaround.

**Review by norabelrose — APPROVED** [review]
LGTM

**Comment by AdamGleave** [comment]
This automation failed after merge for the v0.3.0 tag because our Test PyPI instance had a spurious v0.3.0 package. This problem should go away once we make the next minor release.

**Pull request merged**